PxeReaderSDK/Documentation/PxeReaderUI/man/man3/PxePlayerUIUtil.3
.TH "PxePlayerUIUtil" 3 "Mon Apr 28 2014" "PxeReaderUI" \" -*- nroff -*-
.ad l
.nh
.SH NAME
PxePlayerUIUtil \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#import <PxePlayerUIUtil\&.h>\fP
.PP
Inherits NSObject\&.
.SS "Class Methods"

.in +1c
.ti -1c
.RI "(UILabel *) + \fBcreateLabelWithTitle:frame:\fP"
.br
.ti -1c
.RI "(UILabel *) + \fBcreateLabelWithTitle:frame:alignment:\fP"
.br
.ti -1c
.RI "(UILabel *) + \fBcreateLabelWithTitle:frame:alignment:withFont:backGroundColor:textColor:\fP"
.br
.ti -1c
.RI "(UIBarButtonItem *) + \fBcreateBarButtonItemFixedSpaceWithTarget:action:width:\fP"
.br
.ti -1c
.RI "(UIButton *) + \fBcreateButtonWithBackgroundImage:andText:\fP"
.br
.ti -1c
.RI "(UIButton *) + \fBcreateButtonWithImage:\fP"
.br
.ti -1c
.RI "(UIButton *) + \fBcreateButtonWithFrame:type:\fP"
.br
.ti -1c
.RI "(UIButton *) + \fBcreateButtonWithFrame:title:type:\fP"
.br
.ti -1c
.RI "(UITextField *) + \fBcreateTextFieldWithFrame:text:placeHolder:borderStyle:\fP"
.br
.ti -1c
.RI "(UITextField *) + \fBcreateTextFieldWithFrame:text:placeHolder:borderStyle:secure:\fP"
.br
.ti -1c
.RI "(UITextField *) + \fBcreateTextFieldWithFrame:text:placeHolder:borderStyle:autoCorrection:autoCapitalization:secure:\fP"
.br
.ti -1c
.RI "(UIImageView *) + \fBcreateImageViewWithFileName:\fP"
.br
.ti -1c
.RI "(UIImageView *) + \fBcreateImageViewWithFileName:frame:contentMode:\fP"
.br
.ti -1c
.RI "(UIImageView *) + \fBcreateImageViewWithImage:frame:backgroundColor:contentMode:\fP"
.br
.ti -1c
.RI "(UIImage *) + \fBgetLine:withColor:\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 13 of file PxePlayerUIUtil\&.h\&.
.SH "Method Documentation"
.PP 
.SS "+ (UIBarButtonItem *) createBarButtonItemFixedSpaceWithTarget: (id)target(SEL)action(CGFloat)width"

.PP
Definition at line 45 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UIButton *) createButtonWithBackgroundImage: (NSString*)imageName(NSString*)text"

.PP
Definition at line 52 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UIButton *) createButtonWithFrame: (CGRect)frame(NSString*)title(UIButtonType)type"

.PP
Definition at line 79 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UIButton *) createButtonWithFrame: (CGRect)frame(UIButtonType)type"

.PP
Definition at line 74 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UIButton *) createButtonWithImage: (UIImage*)image"

.PP
Definition at line 63 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UIImageView *) createImageViewWithFileName: (NSString*)fileName"

.PP
Definition at line 112 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UIImageView *) \fBcreateImageViewWithFileName:\fP (NSString*)fileName(CGRect)frame(UIViewContentMode)contentMode"

.PP
Definition at line 118 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UIImageView *) createImageViewWithImage: (UIImage*)image(CGRect)frame(UIColor*)p_color(UIViewContentMode)contentMode"

.PP
Definition at line 125 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UILabel *) createLabelWithTitle: (NSString*)title(CGRect)frame"

.PP
Definition at line 19 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UILabel *) createLabelWithTitle: (NSString*)title(CGRect)frame(NSTextAlignment)alignment"

.PP
Definition at line 26 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UILabel *) createLabelWithTitle: (NSString*)title(CGRect)frame(NSTextAlignment)alignment(UIFont*)font(UIColor*)bgColor(UIColor*)textColor"

.PP
Definition at line 33 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UITextField *) createTextFieldWithFrame: (CGRect)frame(NSString*)text(NSString*)placeHolder(UITextBorderStyle)borderStyle"

.PP
Definition at line 89 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UITextField *) createTextFieldWithFrame: (CGRect)frame(NSString*)text(NSString*)placeHolder(UITextBorderStyle)borderStyle(NSInteger)autoCorrection(NSInteger)autoCapitalization(BOOL)isSecure"

.PP
Definition at line 99 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UITextField *) createTextFieldWithFrame: (CGRect)frame(NSString*)text(NSString*)placeHolder(UITextBorderStyle)borderStyle(BOOL)isSecure"

.PP
Definition at line 94 of file PxePlayerUIUtil\&.m\&.
.SS "+ (UIImage *) getLine: (CGSize)size(CGColorRef)color"

.PP
Definition at line 135 of file PxePlayerUIUtil\&.m\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for PxeReaderUI from the source code\&.


PxeReaderSDK/PxePlayerApiUI/PxePlayerUIUtil.m
//
//  NTUtil.m
//  NTApi
//
//  Created by Satyanarayana on 28/06/13.
//  Copyright (c) 2013 Pearson. All rights reserved.
//


#import "PxePlayerUIUtil.h"
#import "PxePlayerUIConstants.h"

#define kDefaultFont [UIFont systemFontOfSize:12]
#define kDefaultBgColor [UIColor clearColor]
#define kDefaultTextColor [UIColor blackColor]

@implementation PxePlayerUIUtil

+(UILabel*)createLabelWithTitle:(NSString*)title frame:(CGRect)frame {
    
    return [self createLabelWithTitle:title frame:frame alignment:NSTextAlignmentLeft
                             withFont:kDefaultFont backGroundColor:kDefaultBgColor
                            textColor:kDefaultTextColor];
}

+(UILabel*)createLabelWithTitle:(NSString*)title frame:(CGRect)frame alignment:(NSTextAlignment)alignment {
    
    return [self createLabelWithTitle:title frame:frame alignment:alignment
                             withFont:[UIFont systemFontOfSize:12] backGroundColor:[UIColor clearColor]
                            textColor:[UIColor blackColor]];    
}

+(UILabel*)createLabelWithTitle:(NSString*)title frame:(CGRect)frame alignment:(NSTextAlignment)alignment
                       withFont:(UIFont*)font backGroundColor:(UIColor*)bgColor textColor:(UIColor*)textColor {
    
    UILabel* label = [[UILabel alloc] initWithFrame:frame];
    label.backgroundColor = bgColor;
    label.textColor = textColor;
    label.textAlignment = alignment;
    label.text = title;
    label.font = font;
    return label;
}

+(UIBarButtonItem*)createBarButtonItemFixedSpaceWithTarget:(id)target action:(SEL)action width:(CGFloat)width{
    
    UIBarButtonItem *barItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFixedSpace target:target action:action];
    barItem.width = width;
    return barItem;
}

+(UIButton*)createButtonWithBackgroundImage:(NSString*)imageName andText:(NSString*)text {
    UIImage* image = [UIImage imageNamed:BUNDLE_FILE(imageName)];
    CGFloat width = image.size.width;
    CGFloat height = image.size.height;
    
    UIButton* btn = [self createButtonWithFrame:CGRectMake(0.0f, 0.0f, width, height) title:text type:UIButtonTypeCustom];
    [btn setBackgroundImage:image forState:UIControlStateNormal];
    
    return btn;
}

+(UIButton*)createButtonWithImage:(UIImage*)image
{
    CGFloat width = image.size.width;
    CGFloat height = image.size.height;
    
    UIButton *button = [self createButtonWithFrame:CGRectMake(0.0f, 0.0f, width, height) type:UIButtonTypeCustom];
    [button setImage:image forState:UIControlStateNormal];
    
    return button;
}

+(UIButton*)createButtonWithFrame:(CGRect)frame type:(UIButtonType)type
{
    return [self createButtonWithFrame:frame title:@"" type:type];
}

+(UIButton*)createButtonWithFrame:(CGRect)frame title:(NSString*)title type:(UIButtonType)type
{
    UIButton *button = [UIButton buttonWithType:type];
    [button setFrame:frame];
    [button setTitle:title forState:UIControlStateNormal];
    
    return button;
}


+(UITextField*)createTextFieldWithFrame:(CGRect)frame text:(NSString*)text placeHolder:(NSString*)placeHolder borderStyle:(UITextBorderStyle)borderStyle
{
    return [self createTextFieldWithFrame:frame text:text placeHolder:placeHolder borderStyle:borderStyle secure:NO];
}

+(UITextField*)createTextFieldWithFrame:(CGRect)frame text:(NSString*)text placeHolder:(NSString*)placeHolder borderStyle:(UITextBorderStyle)borderStyle secure:(BOOL)isSecure
{
    return [self createTextFieldWithFrame:frame text:text placeHolder:placeHolder borderStyle:borderStyle autoCorrection:UITextAutocorrectionTypeNo autoCapitalization:UITextAutocapitalizationTypeNone secure:isSecure];
}

+(UITextField*)createTextFieldWithFrame:(CGRect)frame text:(NSString*)text placeHolder:(NSString*)placeHolder borderStyle:(UITextBorderStyle)borderStyle autoCorrection:(NSInteger)autoCorrection autoCapitalization:(NSInteger)autoCapitalization secure:(BOOL)isSecure
{
    UITextField *textField = [[UITextField alloc] initWithFrame:frame];
    textField.text = text;
    textField.placeholder = placeHolder;
    textField.borderStyle = borderStyle;
    textField.autocorrectionType = autoCorrection;
    textField.autocapitalizationType = autoCapitalization;
    textField.secureTextEntry = isSecure;
    
    return textField;
}

+(UIImageView*)createImageViewWithFileName:(NSString*)fileName
{
    UIImage *image = [UIImage imageNamed:BUNDLE_FILE(fileName)];
    return [[UIImageView alloc] initWithImage:image];    
}

+(UIImageView*)createImageViewWithFileName:(NSString*)fileName frame:(CGRect)frame contentMode:(UIViewContentMode)contentMode
{
    UIImage *image = [UIImage imageNamed:BUNDLE_FILE(fileName)];
    
    return [self createImageViewWithImage:image frame:frame backgroundColor:[UIColor clearColor] contentMode:contentMode];
}

+(UIImageView*)createImageViewWithImage:(UIImage*)image frame:(CGRect)frame backgroundColor:(UIColor*)p_color contentMode:(UIViewContentMode)contentMode
{
    UIImageView *imageView = [[UIImageView alloc] initWithImage:image];
    [imageView setFrame:frame];
    [imageView setBackgroundColor:p_color];
    [imageView setContentMode:contentMode];
    
    return imageView;
}

+(UIImage*)getLine:(CGSize)size withColor:(CGColorRef)color
{
    UIImage *lineImage = nil;
    UIGraphicsBeginImageContext(size);
    CGContextRef context = UIGraphicsGetCurrentContext();
    CGContextSetStrokeColorWithColor(context, color);
    CGContextSetLineWidth(context, size.height * 0.5f);
    CGContextMoveToPoint(context, 0.0f, size.height * 0.25f);
    CGContextAddLineToPoint(context, size.width, 0.0f);
    CGContextStrokePath(context);
    lineImage = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return lineImage;
}


@end
